Automattic/wp-calypso
# Reader: remove Redux state for tag images

## Proposed Changes

Remove Redux plumbing related to Reader tag images, which are no longer used.

The tag pages now use a simpler header and the tag image is no longer used as of https://github.com/Automattic/wp-calypso/pull/75562.

## Testing Instructions

Visit a new Reader tag page at http://calypso.localhost:3000/tag/mountains and ensure nothing explodes.

Double check to confirm that the query component and selectors are no longer used in the codebase.

## Pre-merge Checklist
